### PR TITLE
fix(std/testing): sanitize escape sequences from assertion errors

### PR DESCRIPTION
**Pull request by caspervonb**

### CODE REVIEW

File: std/testing/asserts.ts [hunk 1]
```diff
@@ -21,15 +21,17 @@ export class AssertionError extends Error {
 
 export function _format(v: unknown): string {
   let string = globalThis.Deno
-    ? stripColor(Deno.inspect(v, {
+    ? Deno.inspect(v, {
```
Comment (nayeemrmn) [inline std/testing/asserts.ts:24]: The intention of `stripColor()` here is to strip the colour that's added stylistically by `Deno.inspect()` based on types. This change will make regular objects look like this:
```
> deno eval 'console.log(Deno.inspect([123456789, 123456789, 123456789, 123456789, 123456789, 123456789, 123456789]).replaceAll("\x1b", "<esc>"))'
[
  <esc>[33m123456789<esc>[39m,
  <esc>[33m123456789<esc>[39m,
  <esc>[33m123456789<esc>[39m,
  <esc>[33m123456789<esc>[39m,
  <esc>[33m123456789<esc>[39m,
  <esc>[33m123456789<esc>[39m,
  <esc>[33m123456789<esc>[39m
]
```

**Review by nayeemrmn — CHANGES_REQUESTED** [review]
I've been meaning to add `Deno.InspectOptions::color = true`, which can be set to false here, stopping `Deno.inspect()` from adding stylistic colour while preserving colour in the inspected value (so only they can be escaped like in this PR). We need that before we can apply this fix.

File: std/testing/asserts.ts (hunk 1 above)
Comment (caspervonb) [reply std/testing/asserts.ts:24]: > his change will make regular objects look

Currently only sanitizing string inputs however.

Comment (caspervonb) [reply std/testing/asserts.ts:24]: But actually; this should be handled at a lower level so I'll end up reverting that.

**Comment by caspervonb** [comment]
This should be fixed in inspect instead.

**Pull request closed without merging**